js/analyzer.js

```diff
@@ -1,65 +1,232 @@
 /**
- * Created by danny on 3/19/15.
+ * Text Analyzer, accepts args for config and
+ * @param args
+ * @constructor
  */
+Analyzer = function (args){
+    this.config = args;
+    this.init();
+};
 
-/*text preprocessor*/
+/**
+ * Analyzer initialisation. Loads defaults and overloads custom settings.
+ */
+Analyzer.prototype.init = function() {
 
+    //init preprocessor if not exists
+    if (typeof yst_pp != 'object' || yst_pp.inputText != this.config.textString) {
```

Comment:
Can we do strict comparisons (probably CodeClimate will yell about this in the future) ?